examples/coins/sources/private_coin.move

```diff
@@ -0,0 +1,44 @@
+module coins::private_coin {
+
+    use std::error;
+    use std::string;
+    use moveos_std::signer;
+    use moveos_std::storage_context::StorageContext;
+    use rooch_framework::coin;
+
+    const ErrorTransferAmountTooLarge: u64 = 1;
+
+    /// This Coin has no `store` ability, 
```

Comment:
What are the usage scenarios of private coin?

Reply:
1. For some organizations internal usage, only available to an allowlist users.
2. Like share in a company，at the beginning, don't allow to transfer arbitrary accounts. When the share becomes public, the coin can be upgraded to public.
3. Other scenarios, the developers want to custom the transfer logic.